crates/autopilot/src/run.rs

```diff
@@ -498,6 +446,58 @@ pub async fn run(args: Arguments) {
     let market_makable_token_list =
         AutoUpdatingTokenList::from_configuration(market_makable_token_list_configuration).await;
 
+    let mut maintenance = Maintenance::new(
+        solvable_orders_cache.clone(),
+        settlement_event_indexer,
+        db.clone(),
+    );
+
+    if let Some(ethflow_contract) = args.ethflow_contract {
+        let start_block = determine_ethflow_indexing_start(
+            &skip_event_sync_start,
+            args.ethflow_indexing_start,
+            &web3,
+            chain_id,
+        )
+        .await;
+
+        // return this
+        let refund_event_handler = EventUpdater::new_skip_blocks_before(
+            // This cares only about ethflow refund events because all the other ethflow
+            // events are already indexed by the OnchainOrderParser.
+            EthFlowRefundRetriever::new(web3.clone(), ethflow_contract),
+            db.clone(),
+            block_retriever.clone(),
+            start_block,
+        )
+        .await
+        .unwrap();
+
+        let custom_ethflow_order_parser = EthFlowOnchainOrderParser {};
+        let onchain_order_event_parser = OnchainOrderParser::new(
+            db.clone(),
+            web3.clone(),
+            quoter.clone(),
+            Box::new(custom_ethflow_order_parser),
+            DomainSeparator::new(chain_id, eth.contracts().settlement().address()),
+            eth.contracts().settlement().address(),
+        );
+
+        // return this
+        let onchain_order_indexer = EventUpdater::new_skip_blocks_before(
+            // The events from the ethflow contract are read with the more generic contract
+            // interface called CoWSwapOnchainOrders.
+            CoWSwapOnchainOrdersContract::new(web3.clone(), ethflow_contract),
+            onchain_order_event_parser,
+            block_retriever,
+            start_block,
+        )
+        .await
+        .expect("Should be able to initialize event updater. Database read issues?");
+
+        maintenance.with_ethflow(onchain_order_indexer, refund_event_handler);
```

Comment:
Mostly moved down this block and instead of pushing these structs into the generic `maintainers` list we now add it to the `Maintenance` struct.